ocaml/libs/http-lib/http_svr.ml

```diff
@@ -359,6 +359,12 @@ let request_of_bio_exn ~proxy_seen ~read_timeout ~total_timeout ~max_length bio
     proxy |> Option.fold ~none:[] ~some:(fun p -> [("STUNNEL_PROXY", p)])
   in
   let open Http.Request in
+  (* Below transformation only keeps one value per key, whereas
+     a fully compliant implementation following Uri's interface
+     would operate on list of values for each key instead *)
+  let kvlist_flatten ls =
+    List.map (function k, v :: _ -> (k, v) | k, [] -> (k, "")) ls
```

Comment:
Interestingly the URI RFC (or in fact I wasn't able to find another RFC that would) doesn't standardise the format of query strings, so it is entirely up to the application how it parses it, and we can choose to drop duplicates here.

We rarely use query strings in XAPI anyway, perhaps the RRD export/VNC console would use them.

Reply:
We do use them for the session_id, task_id etc on several HTTP handlers.